internal/subscription/validation.go

```diff
@@ -185,6 +192,22 @@ func validateSubscriptionName(ctx context.Context, m *Model, req *metrov1.Subscr
 	return nil
 }
 
+func validateFilterExpression(ctx context.Context, m *Model, req *metrov1.Subscription) error {
+	if req.Filter != "" {
+		// validate that the filter expression is in the expected format.
+		// Also convert the expression into a GO Struct
+		f := &Filter{}
+		err := filter.Parser.ParseString(m.Name, req.Filter, f)
```

Comment:
what's the role of passing `m.Name` here?

Reply:
Parser expects fileName in this argument. Given that we use string as an input and fileName is not relevant in our use case, have used subscription name instead for context.

Reply:
But what's the impact? Does it create a file locally? Does it read a file? If it creates file(s) then we need to think of space usage/cleanup.